lib/modules/manager/poetry/extract.ts

```diff
@@ -79,14 +81,14 @@ function extractFromSection(
       }
     }
     const dep: PackageDependency = {
-      depName,
+      depName: depNameNormalized,
```

Comment:
Should the normalized name be used in lookups to e.g. pypi as well?

Reply:
According to PEP 503, yes:

> Below the root URL is another URL for each individual project contained within a repository. The format of this URL is /<project>/ where the <project> is replaced by the normalized name for that project

In practice, using the above example, navigating to https://pypi.org/simple/prometheus_client/ results in a redirect to https://pypi.org/simple/prometheus-client/

Now that I am looking at the documentation again, I see that the project name should also be lowercased:

> The name should be lowercased with all runs of the characters ., -, or _ replaced with a single - character.

Will update the PR to lowercase the normalized version as well.

Reply:
I don't have a strong opinion here. I went with changing the depName to the normalized one to achieve consistency on a name that is functionally correct for dependency management.

Reply:
i would prefer to leave depName as is an update packageName instead.

we already do that in other managers too. so it's consistent inside renovate

Reply:
Works for me, will have an update soon.